department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-13168: Redux implementation for VHA help page form

Resolves [APPEALS-13168](https://vajira.max.gov/browse/APPEALS-13168)

### Description
Initial implementation of the Help app redux store.

### Acceptance Criteria

### Testing Plan
Note: There currently isn't a UI that uses the store so the only way to really test this is through the redux developer browser plugin or creating a selector in one of the help page components after the store is initialized to print to the console log.
1. Visit the /help url in the local environment. This will be using the HelpRootView component.
2. Open up the developer tools and select the redux view. Verify the store state.
![image](https://user-images.githubusercontent.com/109369527/214444657-853afb99-63af-41fa-9081-1c2499743e36.png)
3. Change a few methods in the HelpController and index erb file and see if it is properly reflected in the store state.
4. One additional test that can be done is to create a user and a MembershipRequest and see if it properly reflected in the store.
```ruby
new_user = FactoryBot.create(:user)
new_user.css_id = "redux test1"
new_user.save
new_request = MembershipRequest.new(requestor: new_user, organization: BusinessLine.find_by(url: 'vha'))
new_request.save
```
Switch to the newly created redux test 1 user in the app and navigate back to the /help page.
Check in the redux dev tool to see if it is properly added to the redux store
5. Try adding `new_user` to an organization to see if that properly reflects in the `userOrganizations` portion of the state:
```ruby
BvaIntake.singleton.add_user(new_user)
```
6. You can also check if the feature toggle state will update properly.  First start by checking the `featureToggles` portion in the store to confirm that it's set to `false`
```ruby
# Check to see if the feature toggle is enabled
FeatureToggle.enabled?(:program_office_team_management)
=> false
# Enable it!
FeatureToggle.enable!(:program_office_team_management)
=> true
# Confirm that the feature toggle is enabled
FeatureToggle.enabled?(:program_office_team_management)
=> true
```
Now check the `featureToggles` portion in the store to confirm that it's set to `true`
